process-engine/consumer_api_client
# ♻️ Refactor socket io handling

**Changes:**

1. Implement the `removeSubscription` UseCase, which allows a user to unsubscribe from a notification.
2. Implement the `disconnectSocket` function into the ExternalAccessor, which allows a user to close a Socket.IO client.
3. Implement `subscribeOnce` flag for all notification Subscriptions. This works pretty much like the `subscribeOnce` UseCase for the EventAggregator.
4. The ExternalAccessor should no longer permit to connect to Socket.IO, if the user does not provide an identity.
5. Refactor all Notification subscription functions so that they return the created subscriptions.
6. Move all Auth-Checks from the accessors to the client to avoid code duplication.

**Issues:**

Part of https://github.com/process-engine/process_engine_runtime/issues/#135
Part of https://github.com/process-engine/process_engine_runtime/issues/#157
Part of https://github.com/process-engine/process_engine_runtime/issues/#206

PR: #32

## How can others test the changes?

- Create a client that uses the external accessor
- Create some subscriptions for notifications
- Trigger these notifications to receive them
- Unsubscribe from a notification
- Trigger the notification again
- Notice that the callback you provided for that notification will no longer get triggered
- Dispose the ExternalAccessor
- The Socket.IO client will have disconnected from the server and is no longer able to receive notifications

## PR-Checklist

Please check the boxes in this list after submitting your PR:
